src/datasets/arrow_dataset.py

```diff
@@ -301,17 +299,24 @@ def _get_output_signature(
                     f"Unrecognized array dtype {np_arrays[0].dtype}. \n"
                     "Nested types and image/audio types are not supported yet."
                 )
-            shapes = [array.shape for array in np_arrays]
-            static_shape = []
-            for dim in range(len(shapes[0])):
-                sizes = set([shape[dim] for shape in shapes])
-                if dim == 0:
-                    static_shape.append(batch_size)
-                    continue
-                if len(sizes) == 1:  # This dimension looks constant
-                    static_shape.append(sizes.pop())
-                else:  # Use None for variable dimensions
-                    static_shape.append(None)
+            if (
+                column in dataset
+                and isinstance(dataset.features[column], Sequence)
+                and dataset.features[column].length != -1
+            ):
+                static_shape = [batch_size, dataset.features[column].length]
```

Comment:
I think this is not needed if we do shape inference. My comment about `Sequence` was just an idea if shape inference was not here, sorry for the confusion

You can remove this part

Reply:
Alright!